src/dev_usb_freebsd.c

```diff
@@ -0,0 +1,312 @@
+/*
+spacenavd - a free software replacement driver for 6dof space-mice.
+Copyright (C) 2007-2019 John Tsiombikas <nuclear@member.fsf.org>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#if defined(__FreeBSD__)
+
+#include "config.h"
+#include <dev/usb/usb_ioctl.h>
+#include <dev/usb/usbhid.h>
+#include <sys/ioctl.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <glob.h>
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <strings.h>
+#include <unistd.h>
+
+#include "spnavd.h"
+#include "client.h"
+#include "dev.h"
+#include "dev_usb.h"
+#include "event.h"
+
+#define MAX_AXES	6
+#define DEF_MINVAL	(-500)
+#define DEF_MAXVAL	500
+#define DEF_RANGE	(DEF_MAXVAL - DEF_MINVAL)
+#define IS_DEV_OPEN(dev) ((dev)->fd >= 0)
+
+static void close_hid(struct device *dev)
+{
+	if(IS_DEV_OPEN(dev)) {
+		dev->set_led(dev, 0);
+		close(dev->fd);
+		dev->fd = -1;
+#if NEED_STUFF
```

Comment:
What is this `NEED_STUFF` macro?

Reply:
I had initially included all the extra stuff in dev in case something else needed it, that turned out not to be the case.  I'll remove those blocks.